cmd/fetch-tsa-certs/fetch_tsa_certs.go

```diff
@@ -106,68 +120,109 @@ func fetchCertificateChain(ctx context.Context, parent, intermediateKMSKey, leaf
 		Value:    timestampExt,
 	}}
 
-	isCa := true
-	// default value of 0 for int32
-	var maxIssuerPathLength int32
-
-	csr := &privatecapb.CreateCertificateRequest{
-		Parent: parent,
-		Certificate: &privatecapb.Certificate{
-			// Default to a very large lifetime - CA Service will truncate the
-			// lifetime to be no longer than the root's lifetime.
-			// 20 years (24 hours * 365 days * 20)
-			Lifetime: durationpb.New(time.Hour * 24 * 365 * 20),
-			CertificateConfig: &privatecapb.Certificate_Config{
-				Config: &privatecapb.CertificateConfig{
-					PublicKey: &privatecapb.PublicKey{
-						Format: privatecapb.PublicKey_PEM,
-						Key:    pemPubKey,
-					},
-					X509Config: &privatecapb.X509Parameters{
-						KeyUsage: &privatecapb.KeyUsage{
-							BaseKeyUsage: &privatecapb.KeyUsage_KeyUsageOptions{
-								CertSign: true,
-								CrlSign:  true,
-							},
+	var certChain []*x509.Certificate
+
+	if root == "" {
+		// Create a self signed signing certificate for parentPubKey
+		parentSn, err := cryptoutils.GenerateSerialNumber()
+		if err != nil {
+			return nil, fmt.Errorf("generating serial number: %w", err)
+		}
+
+		parentSkid, err := cryptoutils.SKID(parentPubKey)
+		if err != nil {
+			return nil, fmt.Errorf("generating SKID hash: %w", err)
+		}
+		now := time.Now()
+		cert := &x509.Certificate{
+			SerialNumber: parentSn,
+			Subject: pkix.Name{
+				CommonName:   "sigstore-tsa-selfsigned",
+				Organization: []string{"sigstore.dev"},
+			},
+			SubjectKeyId:          parentSkid,
+			NotBefore:             now,
+			NotAfter:              now.AddDate(20, 0, 0),
```

Comment:
Should we use intermediateValidity (or renamed to parentValidity)?

Reply:
yes